src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,53 +1,109 @@
 package core.basesyntax;
 
-/**
- * <p>Реалізувати свій ArrayList який імплементує інтерфейс List. Дотриматися основних вимог щодо
- * реалізації ArrayList (default capacity, newCapacity...)</p>
- */
+import java.util.NoSuchElementException;
+import java.util.Objects;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private T[] elementData;
+    private int size;
+    private int capacity;
+
+    public ArrayList() {
+        elementData = (T[]) new Object[DEFAULT_CAPACITY];
+        capacity = DEFAULT_CAPACITY;
+    }
+
+    public ArrayList(int capacity) {
+        this.capacity = capacity;
+        elementData = (T[]) new Object[capacity];
+    }
+
+    public void ensureCapacity(int minCapacity) {
+        if (minCapacity > capacity) {
+            capacity = (minCapacity * 3) / 2 + 1;
+            Object[] oldData = elementData;
+            elementData = (T[]) new Object[capacity];
+            System.arraycopy(oldData, 0, elementData, 0, size - 1);
+        }
+    }
+
+    private boolean checkIndex(int index) {
+        if (index >= size) {
+            throw new ArrayIndexOutOfBoundsException("Wrong index of array");
+        }
+        return true;
+    }
 
     @Override
     public void add(T value) {
-
+        size++;
+        ensureCapacity(size);
+        elementData[size - 1] = value;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index == size) {
+            add(value);
+            return;
+        }
+        checkIndex(index);
+        size++;
+        ensureCapacity(size);
+        System.arraycopy(elementData, index, elementData, index + 1, size - index - 1);
+        elementData[index] = value;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        size += list.size();
+        ensureCapacity(size);
```

Comment:
you have capacity of 10. You should add a another list with the size of 10. Total capacity should be 20, right?

Reply:
In my case it will be checked if capacity >20, if yes - ok, if no - it will be increased according to the formula: (20 * 3) / 2 + 1. 
I don't now how is it implemented in real ArrayList and don't want to see it until this task will be done.  And there were no instructions how should it be done. So I implemented it in this way.